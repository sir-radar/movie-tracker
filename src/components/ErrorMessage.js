import React,{useEffect} from 'react'

const ErrorMessage = ({resetStatus}) => {
  //resets error status to empty
  useEffect(() => {
    setTimeout(()=>{
      resetStatus()
    }, 2000)
  },[resetStatus]);

  return (
    <div className="w-25 bg-danger text-white text-center message">
      Something went wrong
    </div>
  )
}

export default ErrorMessage