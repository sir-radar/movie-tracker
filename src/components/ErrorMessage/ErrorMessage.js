import React,{useEffect} from 'react'

const ErrorMessage = ({resetStatus}) => {
  //resets error status to empty
  useEffect(() => {
    setTimeout(()=>{
      resetStatus()
    }, 2000)
  },[resetStatus]);

  return (
    <div data-testid="error-message" className="w-auto bg-danger p-1 text-white text-center message">
      Something went wrong
    </div>
  )
}

export default ErrorMessage