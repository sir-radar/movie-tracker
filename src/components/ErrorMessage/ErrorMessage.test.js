import React from 'react';
import { render, cleanup, act } from '@testing-library/react';
import ErrorMessage from './ErrorMessage';
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer"

jest.useFakeTimers();

afterEach(cleanup)

const resetStatus = jest.fn();

const renderComponent = (resetStatus) => {
  const div = document.createElement("div");
  return render(<ErrorMessage resetStatus={resetStatus}/>, div)
}

describe('ErrorMessage Component', () => {

  it("should call resetStatus after timing out", () => {
    
    renderComponent(resetStatus)
    // move ahead in time by 100ms
    act(() => {
      jest.advanceTimersByTime(100);
    });
    expect(resetStatus).not.toHaveBeenCalled();
    // and then move ahead by 2 seconds
    act(() => {
      jest.advanceTimersByTime(2000);
    });
    expect(resetStatus).toHaveBeenCalled();

  })

  it("renders page text correctly", () => {
    const { getByTestId } = renderComponent(resetStatus)
    expect(getByTestId('error-message')).toHaveTextContent("Something went wrong")
  })


  it("matches snapshot", () => {
    const tree = renderer.create(<ErrorMessage resetStatus={resetStatus}/>).toJSON();
    expect(tree).toMatchSnapshot();
  })

})