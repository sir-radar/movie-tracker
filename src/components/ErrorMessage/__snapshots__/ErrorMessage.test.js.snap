// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage Component matches snapshot 1`] = `
<div
  className="w-auto bg-danger p-1 text-white text-center message"
  data-testid="error-message"
>
  Something went wrong
</div>
`;
